internal/operator/uninstall.go

```diff
@@ -69,82 +81,116 @@ func (u *Uninstall) Run(ctx context.Context) error {
 	if err := u.config.Client.Get(ctx, catsrcKey, catsrc); err != nil {
 		return fmt.Errorf("get catalog source: %v", err)
 	}
-
-	installPlanKey := types.NamespacedName{
-		Namespace: sub.Status.InstallPlanRef.Namespace,
-		Name:      sub.Status.InstallPlanRef.Name,
+	catsrc.TypeMeta = metav1.TypeMeta{
+		Kind:       v1alpha1.CatalogSourceKind,
+		APIVersion: v1alpha1.CatalogSourceCRDAPIVersion,
 	}
```

Comment:
Setting the TypeMeta is necessary to log the kind when deleting with `u.deleteObjects()`